cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/DiskFailureDetector.java

```diff
@@ -66,82 +61,70 @@ public DiskFailureDetector(KafkaCruiseControlConfig config,
       }
       return true;
     }
     _lastCheckedClusterGeneration = currentClusterGeneration;
 
-    Set<Integer> deadBrokers = _loadMonitor.deadBrokers(MAX_METADATA_WAIT_MS);
+    Set<Integer> deadBrokers = _loadMonitor.deadBrokersWithReplicas(MAX_METADATA_WAIT_MS);
     if (!deadBrokers.isEmpty()) {
       if (LOG.isDebugEnabled()) {
-        LOG.debug("Skipping goal violation detection because there are dead broker in the cluster, dead broker: {}",
-                  deadBrokers);
+        LOG.debug("Skipping disk failure detection because there are dead broker in the cluster, dead broker: {}",
+            deadBrokers);
       }
       return true;
     }
 
-    LoadMonitorTaskRunner.LoadMonitorTaskRunnerState loadMonitorTaskRunnerState = _loadMonitor.taskRunnerState();
-    if (!ViolationUtils.isLoadMonitorReady(loadMonitorTaskRunnerState)) {
-      LOG.info("Skipping bad disk detection because load monitor is in {} state.", loadMonitorTaskRunnerState);
-      return true;
-    }
-
-    ExecutorState.State executorState = _kafkaCruiseControl.state(
-        new OperationProgress(), Collections.singleton(EXECUTOR)).executorState().state();
-    if (executorState != ExecutorState.State.NO_TASK_IN_PROGRESS) {
-      LOG.info("Skipping bad disk detection because the executor is in {} state.", executorState);
-      return true;
-    }
-
-    return false;
+    return shouldSkipAnomalyDetection(_loadMonitor, _kafkaCruiseControl);
   }
 
   @Override
   public void run() {
-    if (shouldSkipBadDiskDetection()) {
-      return;
-    }
-
-    Map<Integer, Map<String, Long>> failedDisksByBroker = new HashMap<>();
-    Set<Integer> aliveBrokers = _loadMonitor.kafkaCluster().nodes().stream().mapToInt(Node::id).boxed().collect(Collectors.toSet());
-    _adminClient.describeLogDirs(aliveBrokers).values().forEach((broker, future) -> {
-      try {
-        future.get(LOGDIR_RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS).forEach((logdir, info) -> {
-        if (info.error != Errors.NONE) {
+    try {
+      if (shouldSkipDiskFailureDetection()) {
+        return;
+      }
+      Map<Integer, Map<String, Long>> failedDisksByBroker = new HashMap<>();
+      Set<Integer> aliveBrokers = _loadMonitor.kafkaCluster().nodes().stream().mapToInt(Node::id).boxed().collect(Collectors.toSet());
+      _adminClient.describeLogDirs(aliveBrokers).values().forEach((broker, future) -> {
+        try {
+          future.get(LOGDIR_RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS).forEach((logdir, info) -> {
+            if (info.error != Errors.NONE) {
               failedDisksByBroker.putIfAbsent(broker, new HashMap<>());
               failedDisksByBroker.get(broker).put(logdir, _time.milliseconds());
             }
           });
         } catch (TimeoutException te) {
-          LOG.warn("Getting logdir information for broker {} timed out.", broker);
+          LOG.warn("Retrieving logdir information for broker {} encounter timeout exception {}.", broker, te);
```

Comment:
Is it possible to put `catch (TimeoutException | InterruptedException | ExecutionException e) {` in a single catch block? We already log `te` so no need to say `timeout exception` explicitly.